src/overmap.cpp

```diff
@@ -162,12 +162,21 @@ bool is_river(const oter_id &ter)
 bool is_ot_type(const std::string &otype, const oter_id &oter)
 {
     const size_t compare_size = otype.size();
-    if (compare_size > oter.size()) {
-        return false;
-    } else {
-        return std::string(oter).compare(0, compare_size, otype ) == 0;
-    }
 
+    if (compare_size <= oter.size()) {
+        const auto oter_str = std::string(oter);
+        if (oter_str.compare(0, compare_size, otype) == 0) {
+            // check if it's a full match
+            if (compare_size == oter.size()) {
+                return true;
+            }
+            // only ok for partial if next char is an underscore
+            if (oter_str.at(compare_size) == '_') {
```

Comment:
Four lines above here (still within the diff context), there's a check for `compare_size == oter.size()`... although should that be `oter_str.size()`, maybe?

Either way, if that exception gets raised, something *truly* exceptional has to have happened, because we wouldn't have gone into this branch if `compare_size` was too long, and we can't get here if `compare_size` is exactly the right length. Therefore, compare_size must be at least 1 smaller than `oter_str`'s length, so the next character must exist.